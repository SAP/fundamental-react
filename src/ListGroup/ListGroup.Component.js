import path from 'path';
import React from 'react';
import { Button, ListGroup } from '../';
import { ComponentPage, Example } from '../_playground';

export const ListGroupComponent = () => {
    return (
        <ComponentPage
            description={`Lists and tables are similar as both usually contain a vertical list of data,
                but lists generally contain basic data and tables tend to hold more complex data.
                If the list is a complex hierarchy, it is best to use a **Tree**.`}
            sourceModulePath={path.join(__dirname, './ListGroup')}
            title='List Group'>

            <Example
                description='A link can be used to allow the user to access more details about the item.'
                title='Simple List'>
                <ListGroup>
                    <ListGroup.Item><a style={{ cursor: 'pointer' }}>List item 1</a></ListGroup.Item>
                    <ListGroup.Item>List item 2</ListGroup.Item>
                    <ListGroup.Item><a style={{ cursor: 'pointer' }}>List item 3</a></ListGroup.Item>
                    <ListGroup.Item>List item 4</ListGroup.Item>
                </ListGroup>
            </Example>

            <Example
                description='The List item can contain quick actions.'
                title='List with Actions'>
                <ListGroup>
                    <ListGroup.Item>List item 1
                        <ListGroup.ItemActions>
                            <Button glyph='edit' option='light' />
                        </ListGroup.ItemActions>
                    </ListGroup.Item>
                    <ListGroup.Item>List item 2
                        <ListGroup.ItemActions>
                            <Button glyph='edit' option='light' />
                        </ListGroup.ItemActions>
                    </ListGroup.Item>
                    <ListGroup.Item>List item 3
                        <ListGroup.ItemActions>
                            <Button glyph='edit' option='light' />
                        </ListGroup.ItemActions>
                    </ListGroup.Item>
                    <ListGroup.Item>List item 4
                        <ListGroup.ItemActions>
                            <Button glyph='edit' option='light' />
                        </ListGroup.ItemActions>
                    </ListGroup.Item>
                </ListGroup>
            </Example>

            <Example
                description='Checkboxes can be include on the left of each line for such purposes as bulk actions.'
                title='List with Checkboxes'>
                <ListGroup>
                    <ListGroup.Item><ListGroup.ItemCheckbox>List item 1</ListGroup.ItemCheckbox></ListGroup.Item>
                    <ListGroup.Item><ListGroup.ItemCheckbox>List item 2</ListGroup.ItemCheckbox></ListGroup.Item>
                    <ListGroup.Item><ListGroup.ItemCheckbox>List item 3</ListGroup.ItemCheckbox></ListGroup.Item>
                    <ListGroup.Item><ListGroup.ItemCheckbox>List item 4</ListGroup.ItemCheckbox></ListGroup.Item>
                </ListGroup>
            </Example>
        </ComponentPage>
    );
};
