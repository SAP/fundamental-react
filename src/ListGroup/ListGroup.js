import PropTypes from 'prop-types';
import React from 'react';

export const ListGroup = ({ children, className, ...props }) => {
    return (
        <ul className={`fd-list-group${className ? ' ' + className : ''}`} {...props}>
            {children}
        </ul>
    );
};

ListGroup.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string
};

export const ListGroupItem = ({ children, className, ...props }) => {
    return (
        <li className={`fd-list-group__item${className ? ' ' + className : ''}`} {...props}>
            {children}
        </li>

    );
};

ListGroupItem.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string
};

export const ListGroupItemActions = ({ children, className, ...props }) => {
    return (
        <span className={`fd-list-group__action${className ? ' ' + className : ''}`} {...props}>
            {children}
        </span>
    );
};

ListGroupItemActions.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string
};

export const ListGroupItemCheckbox = (props) => {
    const { children } = props;
    return (
        <div className='fd-form__item fd-form__item--check'>
            <label className='fd-form__label' htmlFor='CndSd399'>
                <input className='fd-form__control' id='CndSd399'
                    type='checkbox' />
                {children}
            </label>
        </div>
    );
};

ListGroupItemCheckbox.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string
};
