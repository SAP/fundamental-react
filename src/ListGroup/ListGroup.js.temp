import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

export const ListGroup = ({ children, className, ...props }) => {
    const listGroupClasses = classnames(
        'fd-list-group',
        className
    );

    return (
        <ul {...props} className={listGroupClasses}>
            {children}
        </ul>
    );
};
ListGroup.displayName = 'ListGroup';

ListGroup.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string
};

export const ListGroupItem = ({ children, className, ...props }) => {
    const listGroupItemClasses = classnames(
        'fd-list-group__item',
        className
    );

    return (
        <li {...props} className={listGroupItemClasses}>
            {children}
        </li>

    );
};
ListGroupItem.displayName = 'ListGroupItem';

ListGroupItem.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string
};

export const ListGroupItemActions = ({ children, className, ...props }) => {
    const listGroupItemActionsClasses = classnames(
        'fd-list-group__action',
        className
    );

    return (
        <span {...props} className={listGroupItemActionsClasses}>
            {children}
        </span>
    );
};
ListGroupItemActions.displayName = 'ListGroupItemActions';

ListGroupItemActions.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string
};

export const ListGroupItemCheckbox = ({ children, labelProps, inputProps, ...props }) => {
    return (
        <div {...props} className='fd-form__item fd-form__item--check'>
            <label
                {...labelProps}
                className='fd-form__label'
                htmlFor='CndSd399'>
                <input
                    {...inputProps}
                    className='fd-form__control'
                    id='CndSd399'
                    type='checkbox' />
                {children}
            </label>
        </div>
    );
};
ListGroupItemCheckbox.displayName = 'ListGroupItemCheckbox';

ListGroupItemCheckbox.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    inputProps: PropTypes.object,
    labelProps: PropTypes.object
};
