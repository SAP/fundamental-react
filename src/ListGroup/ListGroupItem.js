import classnames from 'classnames';
import ListGroupItemActions from './_ListGroupItemActions';
import ListGroupItemCheckbox from './_ListGroupItemCheckbox';
import PropTypes from 'prop-types';
import React from 'react';

const ListGroupItem = ({ children, className, ...props }) => {
    const listGroupItemClasses = classnames(
        'fd-list-group__item',
        className
    );

    return (
        <li {...props} className={listGroupItemClasses}>
            {children}
        </li>

    );
};

ListGroupItem.displayName = 'ListGroupItem';

ListGroupItem.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string
};

ListGroupItem.Actions = ListGroupItemActions;
ListGroupItem.Checkbox = ListGroupItemCheckbox;

export default ListGroupItem;
