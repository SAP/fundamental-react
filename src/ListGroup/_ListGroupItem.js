import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

const ListGroupItem = ({ children, className, ...props }) => {
    const listGroupItemClasses = classnames(
        'fd-list-group__item',
        className
    );

    return (
        <li {...props} className={listGroupItemClasses}>
            {children}
        </li>

    );
};

ListGroupItem.displayName = 'ListGroup.Item';

ListGroupItem.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string
};

export default ListGroupItem;
