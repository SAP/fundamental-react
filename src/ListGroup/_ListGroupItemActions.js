import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

const ListGroupItemActions = ({ children, className, ...props }) => {
    const listGroupItemActionsClasses = classnames(
        'fd-list-group__action',
        className
    );

    return (
        <span {...props} className={listGroupItemActionsClasses}>
            {children}
        </span>
    );
};

ListGroupItemActions.displayName = 'ListGroup.ItemActions';

ListGroupItemActions.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string
};

export default ListGroupItemActions;
