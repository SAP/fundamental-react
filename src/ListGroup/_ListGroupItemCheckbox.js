import Checkbox from '../Forms/Checkbox';
import PropTypes from 'prop-types';
import React from 'react';
import shortid from '../utils/shortId';

const ListGroupItemCheckbox = ({ children, id, ...props }) => {
    return (
        <Checkbox {...props} id={id ? id : shortid.generate()}
            value={children} />
    );
};

ListGroupItemCheckbox.displayName = 'ListGroup.ItemCheckbox';

ListGroupItemCheckbox.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    id: PropTypes.string
};

export default ListGroupItemCheckbox;
