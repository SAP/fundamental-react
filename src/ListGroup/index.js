export { default as ListGroup } from './ListGroup.js';
