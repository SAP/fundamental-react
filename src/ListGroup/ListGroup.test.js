import { Button } from '../Button/Button';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import { ListGroup, ListGroupItem, ListGroupItemActions, ListGroupItemCheckbox} from './ListGroup';

describe('<ListGroup />', () => {
    const defaultListGroup = (
        <ListGroup className='blue'>
            <ListGroupItem className='blue'>
                <a style={{ cursor: 'pointer' }}>List item 1</a>
            </ListGroupItem>
            <ListGroupItem>List item 2</ListGroupItem>
            <ListGroupItem>
                <a style={{ cursor: 'pointer' }}>List item3</a>
            </ListGroupItem>
            <ListGroupItem>List item 4</ListGroupItem>
            <ListGroupItem>
                List item 1
                <ListGroupItemActions className='blue'>
                    <Button glyph='edit' type='standard' />
                </ListGroupItemActions>
            </ListGroupItem>
        </ListGroup>
    );
    const listGroupAction = (
        <ListGroup>
            <ListGroupItem>
                List item 1
                <ListGroupItemActions>
                    <Button glyph='edit' type='standard' />
                </ListGroupItemActions>
            </ListGroupItem>
        </ListGroup>
    );

    const listGroupCheckbox = (
        <ListGroup>
            <ListGroupItem>
                <ListGroupItemCheckbox>List item 1</ListGroupItemCheckbox>
            </ListGroupItem>
        </ListGroup>
    );

    test('create list group', () => {
        // create default list group
        let component = renderer.create(defaultListGroup);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // list group with action
        component = renderer.create(listGroupAction);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // list group with checkbox
        component = renderer.create(listGroupCheckbox);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the ListGroup component', () => {
            const element = mount(<ListGroup data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

        test('should allow props to be spread to the ListGroupItem component', () => {
            const element = mount(<ListGroupItem data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

        test('should allow props to be spread to the ListGroupItemActions component', () => {
            const element = mount(<ListGroupItemActions data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

        xtest('should allow props to be spread to the ListGroupItemCheckbox component', () => {
            // TODO: placeholder for this test description once that functionality is built
            const element = mount(<ListGroupItemCheckbox data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

        xtest('should allow props to be spread to the ListGroupItemCheckbox component\'s input element', () => {
            // TODO: placeholder for this test description once that functionality is built
        });

        xtest('should allow props to be spread to the ListGroupItemCheckbox component\'s label element', () => {
            // TODO: placeholder for this test description once that functionality is built
        });
    });
});
