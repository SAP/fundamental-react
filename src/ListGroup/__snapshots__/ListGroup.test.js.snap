// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListGroup /> create list group 1`] = `
<ul
  className="fd-list-group blue"
>
  <li
    className="fd-list-group__item blue"
  >
    <a
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      List item 1
    </a>
  </li>
  <li
    className="fd-list-group__item"
  >
    List item 2
  </li>
  <li
    className="fd-list-group__item"
  >
    <a
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      List item3
    </a>
  </li>
  <li
    className="fd-list-group__item"
  >
    List item 4
  </li>
  <li
    className="fd-list-group__item"
  >
    List item 1
    <span
      className="fd-list-group__action blue"
    >
      <button
        className="fd-button fd-button--standard sap-icon--edit"
        disabled={false}
        selected={false}
      />
    </span>
  </li>
</ul>
`;

exports[`<ListGroup /> create list group 2`] = `
<ul
  className="fd-list-group"
>
  <li
    className="fd-list-group__item"
  >
    List item 1
    <span
      className="fd-list-group__action"
    >
      <button
        className="fd-button fd-button--standard sap-icon--edit"
        disabled={false}
        selected={false}
      />
    </span>
  </li>
</ul>
`;

exports[`<ListGroup /> create list group 3`] = `
<ul
  className="fd-list-group"
>
  <li
    className="fd-list-group__item"
  >
    <div
      className="fd-form__item fd-form__item--check"
    >
      <label
        className="fd-form__label"
        htmlFor="CndSd399"
      >
        <input
          className="fd-form__control"
          id="CndSd399"
          type="checkbox"
        />
        List item 1
      </label>
    </div>
  </li>
</ul>
`;
